core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java

```diff
@@ -149,6 +154,10 @@ ScannedConfigDocsItemHolder findInMemoryConfigurationItems() {
                 }
             }
 
+            if (isDeprecated) {
+                continue; // do not include deprecated config items
```

Comment:
Although it makes sense to not add deprecated config items, I wonder if we shouldn't still display them in the docs but in a different style (as in ~~`do not use`~~)

Reply:
Yeah, so I asked Manyanda to do that for now.

We have a ton of deprecated stuff in the datasource config and it will really be very confusing if we include it.

So for now, let's go this way, we can refine later if we see a use for it.

Reply:
Yes, I'd say let's avoid the clutter for now especially when displaying the global extension config.